### PR TITLE
[HCICD-452]Wrong summary data when pull_request_target is the event that triggered the workflow

### PR DESCRIPTION
**Pull request by vijayaraj-c**

### CODE REVIEW

**Comment by cardosov** [comment]
Brought @mayur-hitachivantara and @AdityaHitachi to review this PR since, I believe, this changes might clash with other  work they have done to fix https://hv-eng.atlassian.net/browse/HCICD-450

**Comment by mayur-hitachivantara** [comment]
> Brought @mayur-hitachivantara and @AdityaHitachi to review this PR since, I believe, this changes might clash with other work they have done to fix https://hv-eng.atlassian.net/browse/HCICD-450

Thanks, Vitor.
I think what is done in ticket [HCICD-450](https://hv-eng.atlassian.net/browse/HCICD-450) is correct this ticket just need to add condition on top work done in [HCICD-450](https://hv-eng.atlassian.net/browse/HCICD-450).

**Comment by AdityaHitachi** [comment]
> Brought @mayur-hitachivantara and @AdityaHitachi to review this PR since, I believe, these changes might clash with other work they have done to fix https://hv-eng.atlassian.net/browse/HCICD-450

Yes @cardosov and @mayur-hitachivantara.
@vijayaraj-c  just reverted the changes/conditions to the previous code. This code will now encounter issues when attempting to manual trigger, leading to failures for both scenarios [HCICD-445](https://hv-eng.atlassian.net/browse/HCICD-445 ) & [HCICD-450](https://hv-eng.atlassian.net/browse/HCICD-450 ). 

I believe the recent modifications are not required. Any updates to the code should not disrupt its existing functionality. Only an OR condition for` Pull_request_target` concerning Summary and Slack is needed...

File: action.yaml [hunk 1]
```diff
@@ -284,14 +284,14 @@ runs:
 
             blackduck_message=":arrow_right: Result  :${{ steps.all_icons.outputs.blackduck_scan }}:"
 
-        elif [ "${{ github.event_name }}" == "pull_request" ]; then
+        elif [ "${{ github.event_name }}" == "pull_request" ] || ["${{ github.event_name }}" == "pull_request_target" ]; then
```
Comment (AdityaHitachi) [inline action.yaml:287]: Have you tested this with all event types (Pull Request, Pull Request Target, Manual Trigger, Push)? Whether it is working fine or not, **please paste the links to the test cases here.**

File: action.yaml [hunk 2]
```diff
@@ -222,15 +222,15 @@ runs:
...
+            elif [ "${{ github.event_name }}" == "push" ]; then
               sonar_result+="branch=${GITHUB_REF#refs/heads/}"
             fi
             echo "| [Sonar Scan]($sonar_result) | :${{ steps.all_icons.outputs.sonar_scan }}: |" >> $GITHUB_STEP_SUMMARY
         else
             echo "| Sonar Scan | :${{ steps.all_icons.outputs.sonar_scan }}: |" >> $GITHUB_STEP_SUMMARY
         fi
              
-        if [ "${{ github.event_name }}" != "pull_request" ]; then
+        if [ "${{ github.event_name }}" == "push" ]; then
```
[the remark this answers is not in the document]
Comment (vijayaraj-c) [reply action.yaml]: code updated

File: action.yaml [hunk 3]
```diff
@@ -261,7 +261,7 @@ runs:
         blackduck_result=""
         blackduck_message=""
         branch="${GITHUB_REF#refs/heads/}"
-        if [ "${{ github.event_name }}" != "pull_request" ]; then
+        if [ "${{ github.event_name }}" == "push" ]; then
```
[the remark this answers is not in the document]
Comment (vijayaraj-c) [reply action.yaml]: code updated

**Pull request closed without merging**